topiary-cli/src/io.rs

```diff
@@ -11,11 +11,42 @@ use topiary::{Configuration, Language, SupportedLanguage, TopiaryQuery};
 
 use crate::{
     cli::{AtLeastOneInput, ExactlyOneInput, FromStdin},
-    error::{CLIResult, TopiaryError},
+    error::{CLIError, CLIResult, TopiaryError},
     language::LanguageDefinition,
 };
 
-type QueryPath = PathBuf;
+#[derive(Debug, Clone, Hash)]
+pub enum QueryPath {
+    Path(PathBuf),
+    BuiltIn(String),
+}
+
+impl From<PathBuf> for QueryPath {
+    fn from(path: PathBuf) -> Self {
+        QueryPath::Path(path)
+    }
+}
+
+impl From<&PathBuf> for QueryPath {
+    fn from(path: &PathBuf) -> Self {
+        QueryPath::Path(path.clone())
+    }
+}
+
+impl From<String> for QueryPath {
+    fn from(content: String) -> Self {
+        QueryPath::BuiltIn(content)
+    }
+}
```

Comment:
I wonder if this could be genericised over `AsRef<Path>`? The problem is that would also catch `String`, which is handled differently...